### PR TITLE
Update BGPFilter docs

### PR DESCRIPTION
**Pull request by mazdakn**
Update BGPFilter with two new fields for matching: 
- `source`: Available in Calico master, Enterprise (master, v3.17, v3.18)
  - Calico master: https://github.com/projectcalico/calico/pull/8149
  - Enterprise master: https://github.com/tigera/calico-private/pull/6705
  - Enterprise v3.18: https://github.com/tigera/calico-private/pull/6802
  - Enterprise v3.17: https://github.com/tigera/calico-private/pull/6796
- `interface`: Available in Calico master, Enterprise master
  - Calico master: https://github.com/projectcalico/calico/pull/8175
  - Enterprise master: https://github.com/tigera/calico-private/pull/6655

Related PRs:

 

Product Version(s):

Calico: master
Calico Enterprise: master, v3.17, v3.18 

Issue:

https://tigera.atlassian.net/browse/CORE-9798
https://tigera.atlassian.net/browse/CORE-9403

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: calico-enterprise/reference/resources/felixconfig.mdx [hunk 1]
```diff
@@ -137,8 +137,8 @@ spec:
...
...red interface names on each node, this is a manual process. Felix expect the name to end with either .cali or .calico suffix.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | string                                                                                                  | string                                  | wireguard.cali                                                                                                                                                                                                                                                                                                                                                                                                                                    |
```
Comment (ctauchen) [inline calico-enterprise/reference/resources/felixconfig.mdx]:
```suggestion
| wireguardInterfaceName             | Name of the IPv4 WireGuard interface created by Felix. If you change the name and want to clean up the previously configured interface names on each node, this is a manual process. Felix expects the name to end with either .cali or .calico suffix.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | string                                                                                                  | string                                  | wireguard.cali                                                                                                                                                                                                                                                                                                                                                                                                                                    |
```

File: calico-enterprise/reference/resources/felixconfig.mdx [hunk 2]
```diff
@@ -137,8 +137,8 @@ spec:
...
...onfigured interface names on each node, this is a manual process. Felix expect the name to end with either .cali or .calico suffix.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | string                                                                                                  | string                                  | wg-v6.cali                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (ctauchen) [inline calico-enterprise/reference/resources/felixconfig.mdx]:
```suggestion
| wireguardInterfaceNameV6           | Name of the IPv6 WireGuard interface created by Felix. If you change the name and want to clean up the previously configured interface names on each node, this is a manual process. Felix expects the name to end with either .cali or .calico suffix.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | string                                                                                                  | string                                  | wg-v6.cali                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

File: calico-enterprise/reference/resources/felixconfig.mdx (hunk 1 above)
Comment (ctauchen) [reply calico-enterprise/reference/resources/felixconfig.mdx]: change globally

File: calico-enterprise/reference/resources/felixconfig.mdx (hunk 2 above)
Comment (ctauchen) [reply calico-enterprise/reference/resources/felixconfig.mdx]: change globally

**Review by ctauchen — COMMENTED** [review]
Just a few line edits. 

Would this benefit from a technical review from someone else on your team?

Should this not also go into OSS 3.26? Or perhaps it doesn't belong in CE 3.17?

**Comment by mazdakn** [comment]
@ctauchen we back ported the addition of `source` field to EE v3.18 and v3.17, but not to OSS since it is not needed.

**Comment by mazdakn** [comment]
It's always good to get another review, but this is a small change all about adding two new fields.
At this point only @fasaxc know about it. Hopefully he has time to review this.

File: calico-enterprise/reference/resources/bgpfilter.mdx [hunk 3]
```diff
@@ -71,19 +69,23 @@ spec:
...
-| action        | Action to be taken for this CIDR          | `Accept` or `Reject`               | string |         |
+| Field         | Description                               | Accepted Values                                            | Schema | Default |
+| ------------- | ----------------------------------------- | ---------------------------------------------------------- | ------ | ------- |
+| cidr          | IPv6 range                                | A valid IPv6 CIDR                                          | string |         |
+| matchOperator | Method by which to match candidate routes | `In`, `NotIn`, `Equal`, `NotEqual`                         | string |         |
+| source        | Indicator of the source of route          | `RemotePeers` means any route learned from other BGP peers | string |         |
+| interface     | String to match interface names           | A valid pattern to match interfaces                        | string |         |
+| action        | Action to be taken for this rulw          | `Accept` or `Reject`                                       | string |         |
```
Comment (fasaxc) [inline calico-enterprise/reference/resources/bgpfilter.mdx]:
```suggestion
| action        | Action to be taken for this rule          | `Accept` or `Reject`                                       | string |         |
```

File: calico-enterprise/reference/resources/bgpfilter.mdx [hunk 4]
```diff
@@ -71,19 +69,23 @@ spec:
...
-| matchOperator | Method by which to match candidate routes | `In`, `NotIn`, `Equal`, `NotEqual` | string |         |
-| action        | Action to be taken for this CIDR          | `Accept` or `Reject`               | string |         |
+| Field         | Description                               | Accepted Values                                            | Schema | Default |
+| ------------- | ----------------------------------------- | ---------------------------------------------------------- | ------ | ------- |
+| cidr          | IPv6 range                                | A valid IPv6 CIDR                                          | string |         |
+| matchOperator | Method by which to match candidate routes | `In`, `NotIn`, `Equal`, `NotEqual`                         | string |         |
+| source        | Indicator of the source of route          | `RemotePeers` means any route learned from other BGP peers | string |         |
+| interface     | String to match interface names           | A valid pattern to match interfaces                        | string |         |
```
Comment (fasaxc) [inline calico-enterprise/reference/resources/bgpfilter.mdx]: What is the format of the pattern.   Maybe as simple as saying "`*` can be used as a wildcard"

File: calico-enterprise/reference/resources/bgpfilter.mdx [hunk 5]
```diff
@@ -71,19 +69,23 @@ spec:
...
+| action        | Action to be taken for this rulw          | `Accept` or `Reject`                                       | string |         |
```
Comment (fasaxc) [inline calico-enterprise/reference/resources/bgpfilter.mdx]:
```suggestion
| action        | Action to be taken for this rule          | `Accept` or `Reject`                                       | string |         |
```

**Review by fasaxc — DISMISSED** [review]
Couple of minor points.  I think this covers the spec of the new field but I'm worried that users won't know that they should use these fields in most scenarios to get good behaviour.

I think the feature should be put into OS v3.27 but no need to backport (and code should be merged before adding that to the docs IMO).

**Comment by ctauchen** [comment]
@mazdakn While you're making changes, try fetching from upstream and rebasing. I think your build error may be related to a fix we got in last week.

**Comment by mazdakn** [comment]
@fasaxc thanks for the review. The code is already merged into master which will be part of v3.27.

**Pull request merged**